src/AutoRest.CSharp/build/CodeGeneration.targets

```diff
@@ -32,7 +33,8 @@
 
     <Error Text="Following GitHub URLs do not contain commit hash: @(GithubUrlsWithoutHash) please use permalinks for code generation inputs (see https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files) " Condition="'@(GithubUrlsWithoutHash)' != ''" />
 
-    <Exec Command="node $(AutoRestEntryPoint) --max-memory-size=8192 --skip-csproj --skip-upgrade-check --version=$(AutoRestCoreVersion) $(AutoRestInput) $(AutoRestAdditionalParameters) --use=$(MSBuildThisFileDirectory)../tools/netcoreapp3.1/any/ --output-folder=$(MSBuildProjectDirectory)/Generated --clear-output-folder=true --namespace=$(RootNamespace) --shared-source-folders=&quot;$(AzureCoreSharedCodeDirectory);$(AutoRestSharedCodeDirectory)&quot;" />
+    <Exec Command="$(AutoRestBaseCommand) --output-folder=$(MSBuildProjectDirectory)/Generated --namespace=$(RootNamespace)" Condition="'$(NoDefaultNamespaceAndOutputFolder)' != 'true'"/>
```

Comment:
Is msbuild the correct place to put this configuration? Should it instead be in the autorest.md?

Reply:
I agree that `output-folder` and `namespace`  should be configured by autorest.md. In this PR, I just tried to avoid breaking changes if we simply remove these configurations in msbuild as I saw lots of SDKs do not have `output-folder` or `namespace` configured in their autorest.md while allowing SDKs using the values in autorest.md with the new property.

Reply:
I guess another option would be to move the opt-out into the autorest.md file for Azure.ResourceManager. If that opt-out is provided then we essentially ignore these options on the autorest side. 

Although I'm not an expert on how autorest works here so I'm not sure if that is the optimal approach or not.

Reply:
We'll need to touch autorest (core) as currently in autorest.csharp we just read the final config values from autorest which decides which source to use for the config value. And as a general practice, for command line tools, the priority for config values usually goes like command options > environment variables > config files (it's kind of the easier you can make changes to the values, the higher priority it has). We may not want to break that with some opt-out options for the autorest CLI. Therefore, I prefer to make the change within msbuild.